### PR TITLE
test(fuzz): add FuzzProviderResponses (MB/discogs/wikidata/wikipedia) + FuzzMusicBrainzDiffSnapshot

### LINKED ISSUE #1369: test(fuzz): add FuzzMusicBrainzDiffSnapshot for musicbrainz.parseSnapshot
[mode: direct]
[model: sonnet]
[effort: medium]

## Goal
Add a fuzz test for the MusicBrainz diff-snapshot decoder. MB diff snapshots are persisted to the DB and re-decoded later; corruption (disk error, manual DB edit, downgrade) flows directly back through `Unmarshal`.

## Target
- `internal/provider/musicbrainz/diff.go:175` -- `json.Unmarshal([]byte(s), &items)` inside the diff path

## Suggested seed corpus
- Valid snapshot JSON arrays.
- Snapshots with mixed item types.
- Snapshots with NaN / +Inf numeric fields.
- Snapshots with extremely large element counts.

## Acceptance criteria
- [ ] New `FuzzMusicBrainzDiffSnapshot` in `internal/provider/musicbrainz/`.
- [ ] No panics under 60s fuzz runs.

## Charter
Direct M49 fit. Already-isolated function, single-purpose fuzz. Plan issue A9.

### PR DESCRIPTION
**Pull request by sydlexius**
## Summary
- M49 W3 sub-task 3G. Five fuzz funcs across four provider packages, exercising every `json.Unmarshal` site that decodes upstream provider responses:
  - `FuzzMusicBrainzResponses` -> SearchResponse, MBArtist (2x), MBReleaseGroupSearchResponse -- 10 testdata + 15 inline seeds, 671,135 execs/10s
  - `FuzzMusicBrainzDiffSnapshot` -> `normalizeSliceJSON` at diff.go:175 -- 9 inline seeds, 371,756 execs/10s
  - `FuzzDiscogsResponses` -> SearchResponse, ArtistDetail (2x), ArtistReleasesResponse, MasterRelease -- 5 testdata + 14 inline seeds, 500,148 execs/10s
  - `FuzzWikidataResponses` -> SPARQLResponse, CommonsResponse -- 7 testdata + 13 inline seeds, 1,422,421 execs/10s
  - `FuzzWikipediaResponses` -> sparqlResponse, wbEntityResponse (2x), extractResponse, parseResponse -- 8 testdata + 16 inline seeds, 1,227,713 execs/10s

Per-provider `testdata/*` JSON corpora bootstrap the seeds; crafted edge cases (empty `{}`, deeply nested arrays, wrong-type fields, NaN / +Inf numerics, circular id refs, mixed-item-type snapshots) supplement.

## Package access
Each fuzz file is in `package <provider>` (no `_test` suffix on the package decl) for internal access -- `normalizeSliceJSON` in musicbrainz and all five wikipedia response types are unexported.

## Line-number note
Issue body line citations were 1-5 lines off the current source (e.g. wikidata SPARQLResponse decode is at line 281, not 280). All decoder functions and call sites are present and structurally unchanged.

## Test plan
- [x] `go test ./internal/provider/{musicbrainz,discogs,wikidata,wikipedia}/...`
- [x] Five fuzz runs at 10s each, no panics
- [x] `golangci-lint run` clean across all four provider packages

Closes #1368.
Closes #1369.

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive fuzzing tests for Discogs, MusicBrainz (including snapshot normalization), Wikidata, and Wikipedia integrations.
  * Seeds include real testdata plus crafted edge cases (empty/null/wrong-types/malformed/circular-like inputs) to exercise decoding paths and guard against panics, improving resilience and stability when handling external provider responses.

[Review Change Stack](https://app.coderabbit.ai/change-stack/sydlexius/stillwater/pull/1476)

### CODE REVIEW

File: internal/provider/musicbrainz/musicbrainz_fuzz_test.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+	// Snapshots with extremely large element counts (10 000 items).
+	largeArray := make([]byte, 0, 10*1024)
+	largeArray = append(largeArray, '[')
+	for i := 0; i < 1000; i++ {
+		if i > 0 {
+			largeArray = append(largeArray, ',')
+		}
+		largeArray = append(largeArray, '"', 'a', '"')
+	}
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/provider/musicbrainz/musicbrainz_fuzz_test.go]: Fixed in 776700ab — corrected the loop bound to `10000`, sized the buffer to `10000*4+2` (≈40 KB, matches the actual content), and normalized the comment to "10,000 items". The seed now actually exercises extremely-large element-count snapshots as intended.
